applications/FluidDynamicsApplication/custom_response_functions/velocity_pressure_norm_square_response_function.h

```diff
@@ -0,0 +1,451 @@
+        {
+            "norm_model_part_name": "PLEASE_SPECIFY_MODEL_PART",
+            "velocity_norm_factor": 1.0,
+            "pressure_norm_factor": 1.0,
+            "entities"            : ["elements"]
+        })");
+
+        Settings.ValidateAndAssignDefaults(default_settings);
+
+        mNormModelPartName = Settings["norm_model_part_name"].GetString();
+        mVelocityNormFactor = Settings["velocity_norm_factor"].GetDouble();
+        mPressureNormFactor = Settings["pressure_norm_factor"].GetDouble();
+
+        mIsElements = false;
+        mIsConditions = false;
+        const auto& entities = Settings["entities"].GetStringArray();
+        for (const auto& entity : entities) {
+            if (entity == "elements") {
+                mIsElements = true;
+            } else if (entity == "conditions") {
+                mIsConditions = true;
+            } else {
+                KRATOS_ERROR << "Unsupported entity type provided under "
+                                "\"entities\". Supported entity types are:\n"
+                             << "\t elements\n"
+                             << "\t conditions\n.";
+            }
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    /// Destructor.
+    ~VelocityPressureNormSquareResponseFunction() override
+    {
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Initialize() override
+    {
+        KRATOS_TRY;
+
+        KRATOS_ERROR_IF(!mrModelPart.HasSubModelPart(mNormModelPartName))
+            << mNormModelPartName << " model part not found in "
+            << mrModelPart.Name() << ".\n";
+
+        auto& r_norm_model_part = mrModelPart.GetSubModelPart(mNormModelPartName);
+
+        if (mIsElements && !mIsConditions) {
+            const IndexType number_of_entities =
+                r_norm_model_part.GetCommunicator().GlobalNumberOfElements();
+
+            if (number_of_entities == 0) {
+                KRATOS_WARNING("VelocityPressureNormSquareResponseFunction")
+                    << "Only \"elements\" are chosen as entities "
+                       "and no elements are found in "
+                    << r_norm_model_part.Name() << " model part. Therefore trying to compute response function on conditions.\n";
+                mIsConditions = true;
+                mIsElements = false;
+            }
+        }
+
+        VariableUtils().SetFlag(STRUCTURE, false, mrModelPart.Elements());
+        VariableUtils().SetFlag(STRUCTURE, false, mrModelPart.Conditions());
+
+        IndexType total_entities = 0;
+        if (mIsElements) {
+            total_entities += r_norm_model_part.GetCommunicator().GlobalNumberOfElements();
+            VariableUtils().SetFlag(STRUCTURE, true, r_norm_model_part.Elements());
+        }
+
+        if (mIsConditions) {
+            total_entities += r_norm_model_part.GetCommunicator().GlobalNumberOfConditions();
+            VariableUtils().SetFlag(STRUCTURE, true, r_norm_model_part.Conditions());
+        }
```

Comment:
These are only set in the boundary elements / conditions attached to the structure right? (you're passing the boundary model part no?)

Reply:
At the moment yes, but thie response function is made to work with norm calculation for the whole volume `model_part` as well (which I will use at later time), I know using `STRUCUTRE` flag is not the best thing to do, but I was not sure which to use.